src/MainView.vala

```diff
@@ -127,11 +162,24 @@ public class DateTime.MainView : Gtk.Grid {
         GLib.Settings wingpanel_settings = null;
 
         if (schema == null) {
+            show_date_label.visible = false;
+            show_date_switch.visible = false;
+            show_weekday_label.visible = false;
+            show_weekday_switch.visible = false;
+            show_seconds_label.visible = false;
+            show_seconds_switch.visible = false;
             week_number_label.visible = false;
             week_number_switch.visible = false;
         } else {
             wingpanel_settings = new GLib.Settings ("io.elementary.desktop.wingpanel.datetime");
+            wingpanel_settings.bind ("clock-show-date", show_date_switch, "active", SettingsBindFlags.DEFAULT);
+            wingpanel_settings.bind ("clock-show-weekday", show_weekday_switch, "active", SettingsBindFlags.DEFAULT);
+            wingpanel_settings.bind ("clock-show-seconds", show_seconds_switch, "active", SettingsBindFlags.DEFAULT);
             wingpanel_settings.bind ("show-weeks", week_number_switch, "active", SettingsBindFlags.DEFAULT);
+
+            show_date_switch.notify["active"].connect (() => {
+                show_weekday_switch.sensitive = show_date_switch.active;
+            });
```

Comment:
This could also be a binding right?
```suggestion
            wingpanel_settings.bind ("clock-show-date", show_weekday_switch, "sensitive", SettingsBindFlags.INVERT_BOOLEAN);
```